sdk/core/Azure.Core.TestFramework/README.md

```diff
@@ -402,3 +402,41 @@ To download and unpack all artifacts use the `Download-DevOpsRecordings.ps1` scr
 The `Download-DevOpsRecordings.ps1` would wait for active runs to finish before retrieving artifacts unless `-NoWait` switch is used.
 
 **NOTE:** these scripts require being signed in with Azure CLI (https://docs.microsoft.com/cli/azure/authenticate-azure-cli?view=azure-cli-latest) and access to the internal DevOps project (https://dev.azure.com/azure-sdk/internal/) 
+
+## Note on private/non-virtual fields in your clients (such as _clientDiagnostics)
+
+Some bindings require code on the customized side to access fields that are generated. For example:
+
+```csharp
+    // Generated\SparkSessionClient.cs
+    public partial class SparkSessionClient
+    {
+        private readonly ClientDiagnostics _clientDiagnostics;
+        ...
+    }
+```
+
+```csharp
+    // Customization\SparkSessionClient.cs
+    internal virtual Response<SparkSession> GetSparkSession(int sessionId, bool? detailed = null, CancellationToken cancellationToken = default)
+    {
+        using var scope = _clientDiagnostics.CreateScope("SparkSessionClient.GetSparkSession");
+        ...
+    }
+```
+
+The direct approach of using `_clientDiagnostics` here will fail on tests that use the test framework. The proxy generated at runtime uses the constructor with no parameters and _clientDiagnostics will be `null`.
+
+The correct solution is to use a virtual property, so the testing framework stubs it out:
+
+```csharp
+    internal virtual ClientDiagnostics ClientDiagnostics => _clientDiagnostics;
+```
+
+Then update your client code to use the new virtual property.  (Pro tip: you can do a VS rename on the field from `_clientDiagnostics` to `ClientDiagnostics` and then change just the field/property definitions back to `_clientDiagnostics` to make this quicker.)
+
+If it's an `internal` property, it will also require an `InternalsVisisbleTo` addition in your `AssemblyInfo.cs`:
+
+```csharp
+[assembly: InternalsVisibleTo("DynamicProxyGenAssembly2, PublicKey=0024000004800000940000000602000000240000525341310004000001000100c547cac37abd99c8db225ef2f6c8a3602f3b3606cc9891605d02baa56104f4cfc0734aa39b93bf7852f7d9266654753cc297e7d2edfe0bac1cdcf9f717241550e0a7b191195b7667bb4f64bcb8e2121380fd1d9d46ad2d92d2d15605093924cceaf74c4861eff62abf69b9291ed0a340e113be11e6a7d3113e92484cf7045cc7")]
```

Comment:
What do folks think about adding this to the template: https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/template/Azure.Template/src/properties/AssemblyInfo.cs